pkg/api/rest/update.go

```diff
@@ -85,6 +85,12 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx api.Context, obj, old runtime
 	} else {
 		objectMeta.Namespace = api.NamespaceNone
 	}
+	// Ensure requests cannot update generation
+	oldMeta, err := api.ObjectMetaFor(old)
+	if kerr != nil {
+		return kerr
+	}
+	objectMeta.Generation = oldMeta.Generation
```

Comment:
did you sweep to ensure all rest handlers that are setting Generation are doing so in PrepareForUpdate?